addons/product/pricelist.py

```diff
@@ -252,86 +178,72 @@ def _price_rule_get_multi(self, cr, uid, pricelist, products_by_qty_by_partner,
                     # Ignored - incompatible UoM in context, use default product UoM
                     pass
 
-            price_type = 'standard_price' if pricelist.type == 'purchase' else 'list_price'
+            price_type = 'list_price'
             # if Public user try to access standard price from website sale, need to call _price_get.
             price = product_obj._price_get(cr, uid, [product], price_type, context=context)[product.id] or False
```

Comment:
is it really need to call _price_get method ? _price_get function is a return  value of product.list_price also has stuff to convert price into uom if context has uom value and also convert into currency amount if context has currency value.

and already stuff for currency convert in line https://github.com/odoo-dev/odoo/blob/master-simplify-pricelists-rmu/addons/product/pricelist.py#L215
and uom convert in line https://github.com/odoo-dev/odoo/blob/master-simplify-pricelists-rmu/addons/product/pricelist.py#L246

so no need to call _price_get to perform duplicate operations